File: solutions/python/0374-guess-number-higher-or-lower.py
class Solution:
    def guessNumber(self, n: int) -> int:
        left, right = 1, n 
        while left <= right: 
            mid = left + (right - left) // 2 
            res = guess(mid) 
            if res == 0: 
                return mid 
            elif res < 0: 
                right = mid - 1 
            else: 
                left = mid + 1 
        return -1 

File: solutions/cpp/0374-guess-number-higher-or-lower.cpp
class Solution {
public:
    int guessNumber(int n) {
        int left = 1, right = n;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            int res = guess(mid);
            if (res == 0) {
                return mid;
            } else if (res < 0) {
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        return -1;
    }
};
